### PR TITLE
Add a parser for French Polynesia live generation/consumption

### LINKED ISSUE #2917: Data available for French Polynesia
Hi, first issue here, hope I'm not doing it the wrong way!

Real-time production/consumption data for _most_ of the islands of French Polynesia is available here : [https://www.edt.pf/transition-energetique-innovation](url)
Production = consumption since there isn't any exchange with neighboring countries/dependent territories.
Electricity in French Polynesia is only generated by thermal (oil), hydro and solar plants. There used to be wind turbines on a few islands, and there are projects to conduct some other experimentation, but it's not clear if they currently are in operation.

The data above-mentioned covers the electricity generated by EDT/Engie and its subsidiary Marama Nui on the most populated islands (including Tahiti, Moorea, Huahine, Bora Bora) where at least 75-80% of the population lives :
- [https://www.edt.pf/edt-engie](url)
- [https://www.edt.pf/documents/20126/1180028/EDT-chiffres+2018-10.jpg/10e3828d-6eb3-5c82-ab6b-ede4055cfb74?t=1582934078733imagePreview=1](url).

The production on the remaining islands is generated by domestic stations operated by municipalities. I've mapped most of the stations, even on the remote islands using many different sources, don't know if this can prove useful to estimate the total installed capacity for French Polynesia.

There's also historic data available here with a 10-minute update (XLSX files) : [https://www.service-public.pf/sde/documentation-2/production-delectricite-edt-sur-tahiti/](url). Unfortunately it only covers the production on the island of Tahiti (~2/3 of the population).

I've started to write a parser for the data using guidance from the Wiki, but I doubt the method I used is optimal. I have basic knowledge in coding (but am very interested in learning more), so I was wondering if anyone could mentor me on this one.

Cheers :)

### PR DESCRIPTION
**Pull request by autipial**
I've attempted to create a parser for French Polynesia and to fix #2917. First commit here so I'm not totally sure of what I submit and I guess it's also possible to optimize the code, all suggestions are welcome.

About capacity : hydro & solar data is easy to find - cf. PDF referenced in DATA_SOURCES.md (hydro - p. 37, solar - p. 28)
Regarding thermal generation (oil), the figure is the sum of data from the many different PDF reports all located at the URL also referenced in DATA_SOURCES.md.

### CODE REVIEW

**Review by byronwilliams — COMMENTED** [review]
Looks good to me!

File: config/zones.json [hunk 1]
```diff
@@ -4112,6 +4112,30 @@
...
+        -155.0369766919999,
+        -28.14120859199994
+      ],
+      [
+        -134.44298255099994,
+        -7.450127862999921
+      ]
+    ],
+    "capacity": {
+      "hydro": 49.4,
```
Comment (Manu1400) [inline config/zones.json:4127]: 48.2 MW in 2019 in https://www.service-public.pf/sde/wp-content/uploads/sites/15/2021/02/BEPF-2019-VF_online_mail_light_compressed.pdf page 17

Comment (autipial) [reply config/zones.json:4127]: 48.2 MW is only the capacity on Tahiti.

There's also 1.2 MW installed in the Marquesas Islands (on Hiva Oa, Nuku Hiva and Fatu Iva/Fatu Hiva) and 0.06 MW on Ra’iātea/Raiatea -- see p. 27, and also the map p. 26 ("Puissance hydraulique : 48.3 MW").

**Review by skovhus — COMMENTED** [review]
Thank you for your first contribution! I added a few comments and suggestions.

File: parsers/PF.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            
+    data_dict = json.loads(data_table)
+
+    data = {
+        'zoneKey': zone_key,
+        'production': {
+        'biomass': None,
+        'coal': None,
+        'gas': None,
+        'hydro': 0.0,
```
[the remark this answers is not in the document]
Comment (autipial) [reply parsers/PF.py]: I've updated the parser accordingly with your requests. I'm not sure to have understood correctly when to use properly `None` and `0.0`. Let me know if something's wrong!

[the remark this answers is not in the document]
Comment (skovhus) [reply parsers/PF.py]: Which electricity production types do we for **sure know they do _not_ have**? Those should have a default value of `0.0`. As an example, if we know they never use hydro, then it should be `'hydro': 0.0`.

The rest should be `None`.

So are we completely sure that they never use biomass, coal, gas, nuclear and geothermal? Else we should change those to `None`.

[the remark this answers is not in the document]
Comment (autipial) [reply parsers/PF.py]: Yes. Nuclear, coal, gas and geothermal are not used. There have been recent studies about developing biomass plants but so far no capacity has been set up.

File: parsers/PF.py [hunk 3]
```diff
@@ -85,23 +86,18 @@ def fetch_production(
...
+        'biomass': 0.0,
+        'coal': 0.0,
+        'gas': 0.0,
+        'hydro': None,
+        'nuclear': 0.0,
+        'oil': None,
+        'solar': None,
+        'wind': None,
+        'geothermal': 0.0,
+        'unknown': 0.0
```
Comment (Kongkille) [inline parsers/PF.py:100]: I would argue this should also be None, as we don't know for sure whether there is no unknown

Comment (autipial) [reply parsers/PF.py]: Yes, it makes sense. I've updated the parser.

**Review by gwpicard — APPROVED** [review]
Looks good!

**Pull request merged**